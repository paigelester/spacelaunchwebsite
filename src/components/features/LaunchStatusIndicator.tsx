import React from "react";
import { Badge } from "react-bootstrap";

interface LaunchStatusIndicatorProps {
    status: string;
}

const getLaunchStatusVariant = (status: string): string => {
    switch (status) {
        case "Go":
        case "Go for Launch":
        case "Success":
        case "Launch Successful":
            return "success";
        case "TBD":
            return "warning";
        default:
            return "info";
    }
};

const LaunchStatusIndicator = (props: LaunchStatusIndicatorProps) => {
    return (
        <Badge bg={getLaunchStatusVariant(props.status)}>{props.status}</Badge>
    );
};

export default LaunchStatusIndicator;