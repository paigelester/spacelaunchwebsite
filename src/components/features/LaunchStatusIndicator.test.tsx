import React from 'react';
import { render } from '@testing-library/react';
import LaunchStatusIndicator from './LaunchStatusIndicator';

it("renders as expected given status is Go", () => {
    // given
    const status: string = "Go";

    // when
    const { container } = render(<LaunchStatusIndicator status={status} />);

    // then
    expect(container).toMatchSnapshot();
});

it("renders as expected given status is Go for Launch", () => {
    // given
    const status: string = "Go for Launch";

    // when
    const { container } = render(<LaunchStatusIndicator status={status} />);

    // then
    expect(container).toMatchSnapshot();
});

it("renders as expected given status is Success", () => {
    // given
    const status: string = "Success";

    // when
    const { container } = render(<LaunchStatusIndicator status={status} />);

    // then
    expect(container).toMatchSnapshot();
});

it("renders as expected given status is Launch Successful", () => {
    // given
    const status: string = "Launch Successful";

    // when
    const { container } = render(<LaunchStatusIndicator status={status} />);

    // then
    expect(container).toMatchSnapshot();
});

it("renders as expected given status is TBD", () => {
    // given
    const status: string = "TBD";

    // when
    const { container } = render(<LaunchStatusIndicator status={status} />);

    // then
    expect(container).toMatchSnapshot();
});

it("renders as expected given status is unknown", () => {
    // given
    const status: string = "unknown";

    // when
    const { container } = render(<LaunchStatusIndicator status={status} />);

    // then
    expect(container).toMatchSnapshot();
});
