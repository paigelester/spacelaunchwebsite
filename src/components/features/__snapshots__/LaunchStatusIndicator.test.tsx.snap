// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected given status is Go 1`] = `
<div>
  <span
    class="badge bg-success"
  >
    Go
  </span>
</div>
`;

exports[`renders as expected given status is Go for Launch 1`] = `
<div>
  <span
    class="badge bg-success"
  >
    Go for Launch
  </span>
</div>
`;

exports[`renders as expected given status is Launch Successful 1`] = `
<div>
  <span
    class="badge bg-success"
  >
    Launch Successful
  </span>
</div>
`;

exports[`renders as expected given status is Success 1`] = `
<div>
  <span
    class="badge bg-success"
  >
    Success
  </span>
</div>
`;

exports[`renders as expected given status is TBD 1`] = `
<div>
  <span
    class="badge bg-warning"
  >
    TBD
  </span>
</div>
`;

exports[`renders as expected given status is unknown 1`] = `
<div>
  <span
    class="badge bg-info"
  >
    unknown
  </span>
</div>
`;
